x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TextFormat.java

```diff
@@ -79,6 +79,9 @@ String format(RestRequest request, SqlQueryResponse response) {
             } else if (response.hasId()) {
                 // an async request has no results yet
                 return StringUtils.EMPTY;
+            } else if (response.rows().isEmpty()) {
+                // no data and no headers to return
+                return StringUtils.EMPTY;
             }
```

Comment:
```suggestion
            } else if (response.hasId() || response.rows().isEmpty()) {
                // for an async request that has no results yet or for an emtpy page, there is nothing to return (no data and no headers)
                return StringUtils.EMPTY;
            }
```

Just integrate this `else if` branch in the previous one, keeping the comment.